### ISSUE TITLE
CHR License Structure

### ISSUE THREAD

**Issue by bimaindrawans** [elseif/MikroTikPatch#44]
I used the license.py file from your previous repository update, but I encountered an issue when generating a license for the CHR version. Would you be willing to share the structure for creating a license for the CHR type? This is because the software ID structure is different from the regular x86, which only has 9 characters, while in CHR there is a combination of letters and symbols totaling 11 characters.

**Comment by israelrmm** [comment]
you must check this

https://t.me/mikrotikpatch/13486

**Comment by bimaindrawans** [comment]
> you must check this
> 
> https://t.me/mikrotikpatch/13486

i didn't find chr structure on that script, because i think the structure written on the license_generator files.

**Comment by israelrmm** [comment]
> you must check this
> 
> https://t.me/mikrotikpatch/13486

if you see that source code. RouterOS Lisecnse Key XXXX-XXXX has 6bit, and "little" is true....  

chr lisense could be 3bit and "little" is false .. (See attchment)

then maybe   lic = software_id.to_bytes(6, 'little')  --->   lic = software_id.to_bytes(3, ???)    ???=big?

and in "micro.py" ( mikro_softwareid_decode()   function )   SOFTWARE_ID_CHARACTER_TABLE = b'TN0BYX18S5HZ4IA67DGF3LPCJQRUK9MW2VE' must be  --->

SOFTWARE_ID_CHARACTER_TABLE = ????????????????????   <- This varible for chr is the trick and we need to found it.

![photo_2024-08-26_23-34-31](https://github.com/user-attachments/assets/de0b9c98-fbea-4485-9aae-7883804f0b0b)

**Comment by vnxme** [comment]
> if you see that source code. RouterOS Lisecnse Key XXXX-XXXX has 6bit, and "little" is true....
> 
> chr lisense could be 3bit and "little" is false .. (See attchment)
> 
> then maybe lic = software_id.to_bytes(6, 'little') ---> lic = software_id.to_bytes(3, ???) ???=big?

No, 6 and 3 have nothing to do with the number of bytes a software ID can take in the license payload. These numbers indicate a license level. For CHR, the third level corresponds to the P-Unlimited.

**Comment by israelrmm** [comment]
> > if you see that source code. RouterOS Lisecnse Key XXXX-XXXX has 6bit, and "little" is true....
> > chr lisense could be 3bit and "little" is false .. (See attchment)
> > then maybe lic = software_id.to_bytes(6, 'little') ---> lic = software_id.to_bytes(3, ???) ???=big?
> 
> No, 6 and 3 have nothing to do with the number of bytes a software ID can take in the license payload. These numbers indicate a license level. For CHR, the third level corresponds to the P-Unlimited.

Ok. i understand. Thanks

**Comment by loskiq** [comment]
parse chr license
https://github.com/loskiq/MikroTikPatch/blob/main/license.py#L29

generate chr license
https://github.com/loskiq/MikroTikPatch/blob/main/license.py#L58
